lisa/tools/go.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import Any, List, Type
+
+from lisa.base_tools.wget import Wget
+from lisa.executable import Tool
+from lisa.operating_system import Posix
+from lisa.tools.echo import Echo
+from lisa.tools.rm import Rm
+from lisa.tools.tar import Tar
+from lisa.util import LisaException
+
+
+class Go(Tool):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        go_version = kwargs.pop("go_version", None)
+        if go_version:
+            self._install_specific_version(go_version)
+
+    @property
+    def command(self) -> str:
+        return "go"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Wget, Tar, Rm]
+
+    def _install(self) -> bool:
+        if isinstance(self.node.os, Posix):
+            self.node.os.install_packages("golang-go")
+        else:
+            raise LisaException(
+                "Doesn't support to install Go in Windows. "
+                "Make sure Go is installed and in PATH"
+            )
+        return self._check_exists()
+
+    def _install_specific_version(self, version: str) -> bool:
+        version_list = []
+        version_list.append("1.15")
```

Comment:
The logic looks awful, and I don't want to add new versions each time. It should directly download the go package. If it doesn't exist, raise the error with friendly message.

Reply:
Sure, i will make change and if any error occurs, I will throw error with proper message.